src/context/mod.rs

```diff
@@ -408,7 +752,49 @@ pub trait ContextSideEffects<'ctx>: private::SealedContext<'ctx> {
 
 impl<'ctx, CTX: private::SealedContext<'ctx>> ContextSideEffects<'ctx> for CTX {}
 
-/// Trait exposing Restate K/V read functionalities.
+/// # State
+/// You can store key-value state in Restate.
+/// Restate makes sure the state is consistent with the processing of the code execution.
+///
+/// **This feature is only available for Virtual Objects and Workflows:**
+/// - For **Virtual Objects**, the state is isolated per Virtual Object and lives forever (across invocations for that object).
+/// - For **Workflows**, you can think of it as if every workflow execution is a new object. So the state is isolated to a single workflow execution. The state can only be mutated by the `run` handler of the workflow. The other handlers can only read the state.
+///
+/// **Info: Command-line introspection**:
```

Comment:
For this type of stuff i don't like bold, just use headers level 3 or 4

```suggestion
/// ### Command-line introspection
```

Reply:
And also remove info/caution/etc

Reply:
and move it to the bottom after the code snippet